R/methods.R

```diff
@@ -300,11 +306,13 @@ residuals.nonprobsvy <- function(object,
     }
     r <- object$selection$residuals
     res_sel <- switch(type,
-                      "pearson" = (R - propensity_scores)/sqrt(propensity_scores * (1 - propensity_scores)),
+                      "response" = r,
+                      "working" = r / propensity_scores * (1 - propensity_scores),
+                      "pearson" = r / sqrt(propensity_scores * (1 - propensity_scores)),
                       "deviance" = s * sqrt(-2 * (R * log(propensity_scores) + (1 - R) * log(1 - propensity_scores))),
-                      "response" = R - propensity_scores,
-                      "pearsonSTD" = r/sqrt( (1 - hatvalues(object)$selection) * object$selection$variance)
-                      ) # TODO studentized_pearson, studentized_deviance
+                      "pearsonSTD" = r/sqrt( (1 - hatvalues(object)$selection) * object$selection$variance),
+                       stop("Invalid type of residual. Choose from 'pearson', 'working', 'deviance', 'response', 'pearsonSTD'.")
```

Comment:
this switch will result in base R error if `type` is `NULL` and will return `NULL` if `type == NA` probably doesn't matter here but its always good to remember that `stop` in `switch` can break if end user tries to break it :)